File: in_Java/0347 Top K Frequent Elements.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        Map<Integer,Integer> hm = new HashMap<>();
        for(int i : nums) hm.put(i, hm.getOrDefault(i,0)+1);
        PriorityQueue<Integer> pq = new PriorityQueue<>((a,b) -> (b-a));
        for(int x : hm.values()) pq.add(x);
        int[] res = new int[k];
        int id = 0;
        int kk = k;
        while(k-- != 0) res[id++] = pq.poll();
        for(int i = 0; i < kk; i++){
            int cur = res[i];
            for(int key : hm.keySet()){
                if(hm.get(key) == cur){
                    res[i] = key;
                    hm.remove(key);
                    break;
                }
            } 
        }
        return res;
    }
}

File: in_Python/0347 Top K Frequent Elements.py
class Solution(object):
    def topKFrequent(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: List[int]
        """
        d = {}
        for i in nums:
            if i not in d:
                d[i] = 1
            else:
                d[i] += 1
        q = []
       # heapq.heapify(q)
        for key in d:
            t = (d[key],key)
            heappush(q,t)
        res = []
        while len(q) > k:
            heappop(q)
        while len(q) != 0:
            g = heappop(q)
            #print(g)
            res.insert(0, g[1])
        return res